src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -822,6 +831,16 @@ private File resolveArtifactSources( Artifact artifact )
                 simpleFilters.add( new SimpleFilter( jars, filter.getIncludes(), filter.getExcludes() ) );
             }
         }
+        else if ( this.filters == null )
+        {
+            getLog().debug( "Adding META-INF/*.SF, META-INF/*.DSA and META-INF/*.RSA exclusions" );
+
+            Map<Artifact, ArtifactId> artifacts = getArtifactIds();
+            simpleFilters.add( new SimpleFilter(
+                    getMatchingJars( artifacts , new ArtifactId( "*:*" ) ),
+                    Collections.<String>emptySet(),
+                    new HashSet<>( Arrays.asList( "META-INF/*.SF", "META-INF/*.DSA", "META-INF/*.RSA" ) ) ) );
+        }
 
         filters.addAll( simpleFilters );
```

Comment:
I would expect a NPE here with the default filters.

Reply:
not sure i'm following, this is not this.filters but the list created in the method